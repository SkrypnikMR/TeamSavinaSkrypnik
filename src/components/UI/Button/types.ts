export interface IStyledButton{
    margin?: string;
    padding?: string;
    width?: string;
    height?: string;
    cursor?: string;
    bgColor: string;
    borderRadius?: string;
    border?: string;
    outline?: string;
    transition?: string;
    fontSize?: string;
    focusColor?: string;
    bgColorDisabled?: string;
    type: 'submit' | 'reset' | 'button' | undefined;
    isDisabled?: boolean;
    fontWeight?: string;
    bgColorHover?: string;
}
export interface IButton {
    id: string;
    name: string;
    onClick: () => void;
    value: string;
    type: 'submit' | 'reset' | 'button' | undefined;
    title: string;
    borderRadius?: string;
    isDisabled?: boolean;
    color?: string;
    fontSize?: string;
    fontWeight?: string;
    bgColor?: string;
    bgColorHover?: string;
    height?: string;
    width?: string;
    padding?: string;
    margin?: string;
    content?: string;
    transition?: string;
    focusColor?: string;
    bgColorDisabled?: string;
}
