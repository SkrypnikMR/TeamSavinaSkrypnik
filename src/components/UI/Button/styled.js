import styled from 'styled-components';
import {
  bgColorDefault,
  colorDefault,
  focusColorDefault,
  disabledColor,
  bgColorDisabledDefault,
  disabledBorder,
} from '../baseLayout';

export const StButton = styled.button`
  ${({ margin = '0 0 3px 0' }) => margin && `margin: ${margin}`};
  font-family: 'Play', sans-serif;
  padding: ${({ padding = '5px' }) => padding};
  width: ${({ width = '170px' }) => width};
  height: ${({ height = '35px' }) => height};
  ${({ cursor = 'pointer' }) => cursor && `cursor: ${cursor}`};
  color: ${({ color = colorDefault }) => color};
  background-color: ${({ bgColor = bgColorDefault }) => bgColor};
  border-radius: ${({ borderRadius = '7px' }) => borderRadius}; 
  border: ${({ border = 'none' }) => border};
  outline: ${({ outline = 'none' }) => outline};
  ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`};
  font-size: ${({ fontSize = '16px' }) => fontSize};
  &:hover {
    color: ${({ focusColor = focusColorDefault }) => focusColor}; 
    border: ${({ border = 'none' }) => border};
    }
  &:focus {
    color: ${({ focusColor = focusColorDefault }) => focusColor};
    }
  &:disabled{ 
    border: ${({ border = disabledBorder }) => border};
    background-color: ${({ bgColorDisabled = bgColorDisabledDefault }) => bgColorDisabled};
    color: ${({ focusColor = disabledColor }) => focusColor};
    }
`;
