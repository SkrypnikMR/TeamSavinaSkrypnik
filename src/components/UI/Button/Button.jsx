import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import { StButton } from './styled';

const Button = ({
    id,
    name,
    onClick,
    title,
    type = 'button',
    value,
    borderRadius,
    isDisabled,
    color,
    fontSize,
    fontWeight,
    bgColorDisabled,
    bgColor,
    bgColorHover,
    height,
    width,
    padding,
    margin,
    content,
    transition,
    focusColor,
}) => {
    const { t } = useTranslation();
    return (
        <StButton
            id={id}
            name={name}
            onClick={onClick}
            type={type}
            value={value}
            title={title}
            bgColorDisabled={bgColorDisabled}
            borderRadius={borderRadius}
            isDisabled={isDisabled}
            color={color}
            fontSize={fontSize}
            fontWeight={fontWeight}
            bgColor={bgColor}
            bgColorHover={bgColorHover}
            height={height}
            width={width}
            padding={padding}
            margin={margin}
            transition={transition}
            focusColor={focusColor}
        >
            {t(content)}
        </StButton>
    );
};

Button.propTypes = {
    id: PropTypes.string.isRequired,
    name: PropTypes.string,
    type: PropTypes.string,
    value: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    title: PropTypes.string,
    borderRadius: PropTypes.string,
    isDisabled: PropTypes.string,
    color: PropTypes.string,
    fontSize: PropTypes.string,
    fontWeight: PropTypes.string,
    bgColor: PropTypes.string,
    bgColorHover: PropTypes.string,
    height: PropTypes.string,
    width: PropTypes.string,
    padding: PropTypes.string,
    margin: PropTypes.string,
    content: PropTypes.string,
    transition: PropTypes.string,
    focusColor: PropTypes.string,
    bgColorDisabled: PropTypes.string,
};

export default Button;
