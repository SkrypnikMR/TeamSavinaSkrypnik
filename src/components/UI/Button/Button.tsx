import React from 'react';
import { useTranslation } from 'react-i18next';
import { StButton } from './styled';
import { useTheme } from '../../Hook/useTheme';
import { IButton } from './types';

const Button = ({
    id,
    name,
    onClick,
    title,
    type = 'button',
    value,
    borderRadius,
    isDisabled,
    color,
    fontSize,
    fontWeight,
    bgColorDisabled,
    bgColor,
    bgColorHover,
    height,
    width,
    padding,
    margin,
    content,
    transition,
    focusColor,
    position,
    right,
} : IButton) => {
    const { t } = useTranslation();
    const { colors, theme } = useTheme();

    return (
        <StButton
            id={id}
            name={name}
            onClick={onClick}
            type={type}
            value={value}
            title={title}
            bgColorDisabled={bgColorDisabled}
            borderRadius={borderRadius}
            isDisabled={isDisabled}
            color={color}
            fontSize={fontSize}
            fontWeight={fontWeight}
            bgColor={bgColor}
            bgColorHover={bgColorHover}
            height={height}
            width={width}
            padding={padding}
            margin={margin}
            transition={transition}
            focusColor={focusColor}
            colors={colors}
            theme={theme}
            position={position}
            right={right}
        >
            {t(content)}
        </StButton>
    );
};

export default Button;
