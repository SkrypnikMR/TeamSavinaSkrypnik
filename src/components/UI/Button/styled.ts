import styled from 'styled-components';
import { IStyledButton } from './types';
import {
  bgColorDefault,
  colorDefault,
  focusColorDefault,
  disabledColor,
  bgColorDisabledDefault,
  disabledBorder,
  bgColorDefaultFon,
  bgMessage,
} from '../baseLayout';

export const StButton = styled.button < IStyledButton > `
  ${({ margin = '0 0 7px 0' }) => margin && `margin: ${margin}`};
  font-family: 'Play', sans-serif;
  padding: ${({ padding = '5px' }) => padding};
  width: ${({ width = '170px' }) => width};
  height: ${({ height = '35px' }) => height};
  ${({ cursor = 'pointer' }) => cursor && `cursor: ${cursor}`};
  color: ${({ color = colorDefault }) => color};
  background-color: ${({ bgColor = bgColorDefault }) => bgColor};
  border-radius: ${({ borderRadius = '7px' }) => borderRadius}; 
  border: ${({ border = 'none' }) => border};
  outline: ${({ outline = 'none' }) => outline};
  ${({ transition = 'all 300ms ease-in-out' }) => transition && `transition: ${transition}`};
  font-size: ${({ fontSize = '16px' }) => fontSize};
  position: ${({ position = 'static' }) => position};
  right: ${( right = '0px' ) => right};
  &:hover {
    background: ${({ bgColor = bgMessage }) => bgColor};
    color: ${({ colors, theme }) => colors[theme].buttonText};
    border: ${({ border = 'none' }) => border};
    }
  &:focus {
    color: ${({ focusColor = focusColorDefault }) => focusColor};
    }
  &:disabled{ 
    border: ${({ border = disabledBorder }) => border};
    background-color: ${({ bgColorDisabled = bgColorDisabledDefault }) => bgColorDisabled};
    color: ${({ focusColor = disabledColor }) => focusColor};
    }
`;
