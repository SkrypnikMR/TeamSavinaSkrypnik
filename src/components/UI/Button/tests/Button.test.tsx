import React from 'react';
import { shallow } from 'enzyme';
import Button from '../Button';

describe('Button', () => {
    let props;
    beforeEach(() => {
        props = {
            id: 'id',
            onClick: jest.fn(),
            title: 'Button',
            name: 'name',
            type: 'Button',
            value: 'value',
            isDisabled: 'false',
        };
    });
    it('Should match snapshot', () => {
        const component = shallow(<Button {...props}/>);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render StButton', () => {
        const component = shallow(<Button {...props}/>);
        expect(component.find('styled__StButton')).toHaveLength(1);
    });
    it('should call onClick', () => {
        const component = shallow(<Button {...props}/>);
        component.find('styled__StButton').simulate('click');
        expect(props.onClick).toHaveBeenCalled();
    });
});
