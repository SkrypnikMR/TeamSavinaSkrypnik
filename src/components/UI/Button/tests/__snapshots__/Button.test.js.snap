// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match snapshot 1`] = `"<button id=\\"id\\" name=\\"name\\" type=\\"Button\\" value=\\"value\\" title=\\"Button\\" class=\\"styled__StButton-sc-1a6hck8-0 jZiUSN\\"></button>"`;
